File: day16/prime_finder.py
import math

def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True

def print_primes_up_to(n):
    for i in range(2, n + 1):
        if is_prime(i):
            print(i, end=" ")
    print()

def get_user_input():
    while True:
        try:
            user_input = int(input("Enter a positive whole number: "))
            if user_input < 2:
                print("Please enter a number 2 or greater.")
                continue
            return user_input
        except ValueError:
            print("Please enter a valid number.")

def main():
    num = get_user_input()
    print(f"Prime numbers up to {num}:")
    print_primes_up_to(num)

if __name__ == "__main__":
    main()

File: day16/PrimeFinder.java
import java.util.Scanner;

public class PrimeFinder {

    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        return true;
    }

    public static void printPrimesUpTo(int n) {
        for (int i = 2; i <= n; i++) {
            if (isPrime(i)) {
                System.out.print(i + " ");
            }
        }
        System.out.println(); // New line after printing all primes
    }

    public static int getUserInput() {
        Scanner scanner = new Scanner(System.in);
        int num;
        while (true) {
            System.out.print("Enter a positive whole number: ");
            String input = scanner.nextLine();
            try {
                num = Integer.parseInt(input);
                if (num >= 2) {
                    break;
                } else {
                    System.out.println("Please enter a number 2 or greater.");
                }
            } catch (NumberFormatException e) {
                System.out.println("Please enter a valid number.");
            }
        }
        scanner.close();
        return num;
    }

    public static void main(String[] args) {
        int userNum = getUserInput();
        System.out.println("Prime numbers up to " + userNum + ":");
        printPrimesUpTo(userNum);
    }
}
